server/settings_test.go

```diff
@@ -618,20 +622,80 @@ groups:
 
 					res, err := serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
 						Body: server.ChangeSettingsBody{
-							AlertManagerURL:   url,
+							AlertManagerURL:   alertManagerURL,
 							AlertManagerRules: rules,
 						},
 						Context: pmmapitests.Context,
 					})
 					require.NoError(t, err)
-					assert.Equal(t, url, res.Payload.Settings.AlertManagerURL)
+					assert.Equal(t, alertManagerURL, res.Payload.Settings.AlertManagerURL)
 					assert.Equal(t, rules, res.Payload.Settings.AlertManagerRules)
 
 					gets, err := serverClient.Default.Server.GetSettings(nil)
 					require.NoError(t, err)
-					assert.Equal(t, url, gets.Payload.Settings.AlertManagerURL)
+					assert.Equal(t, alertManagerURL, gets.Payload.Settings.AlertManagerURL)
 					assert.Equal(t, rules, gets.Payload.Settings.AlertManagerRules)
 
+					t.Run("PrometheusReloaded", func(t *testing.T) {
+						prometheusBaseURL := pmmapitests.BaseURL.ResolveReference(&url.URL{
+							Path: "prometheus/",
+						})
+						client, err := prometheusApi.NewClient(prometheusApi.Config{
+							Address: prometheusBaseURL.String(),
+						})
+						require.NoError(t, err)
+
+						expected := prometheusApiV1.RuleGroup{
+							Rules: prometheusApiV1.Rules{
+								prometheusApiV1.AlertingRule{
+									Alerts:      []*prometheusApiV1.Alert{},
+									Annotations: model.LabelSet{"summary": "High request latency"},
+									Duration:    600,
+									Labels:      model.LabelSet{"severity": "page"},
+									Name:        "HighRequestLatency",
+									Query:       `job:request_latency_seconds:mean5m{job="myjob"} > 0.5`,
+								},
+							},
+							File:     "/srv/prometheus/rules/pmm.rules.yml",
+							Interval: 60,
+							Name:     "example",
+						}
+
+						// Prometheus needs some time to reload configs.
+						ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+						defer cancel()
+
+						api := prometheusApiV1.NewAPI(client)
+						waitCount := 0
+						for {
+							result, err := api.Rules(ctx)
+							// Health is not reliable. It might return ok but many times it returns unknown even after a lot of retries
```

Comment:
I think we should understand the issue better. Why health is not reliable Is it Prometheus bug? We should not add workarounds until we understand it.

Reply:
Health might be not reliable because of a few reasons:
1. We are reloading prometheus configs asynchronously
2. Prometheus may contain more checks in their healthcheck than just checking that new files loaded.

Reply:
That doesn't explain "even after a lot of retries" part

Reply:
Do we wait between retries? It might take a few seconds until everything is ready.

Reply:
time.Sleep(time.Second) is just below

Reply:
There is better question: Will we have the same problem if we will write test for checking that prometheus config is reloaded when we add agent?